src/components/CodeSnippet/CodeSnippet.tsx

```diff
@@ -0,0 +1,96 @@
+import Prism from 'prismjs'
+import 'prismjs/components/prism-bash'
+import 'prismjs/components/prism-javascript'
+import 'prismjs/components/prism-json'
+import 'prismjs/components/prism-ruby'
+import 'prismjs/plugins/line-numbers/prism-line-numbers'
+import { memo, useEffect, useRef } from 'react'
+
+import { Button, Typography } from '~/components/designSystem'
+import { addToast } from '~/core/apolloClient'
+import { copyToClipboard } from '~/core/utils/copyToClipboard'
+import { useInternationalization } from '~/hooks/core/useInternationalization'
+import { tw } from '~/styles/utils'
+
+import './codeSnippet.css'
+
+Prism.manual = true
+
+interface CodeSnippetProps {
+  className?: string
+  loading?: boolean
+  code: string
+  language?: 'bash' | 'javascript' | 'json'
+  canCopy?: boolean
+  displayHead?: boolean
+}
+
+export const CodeSnippet = memo(
+  ({
+    className,
+    loading,
+    code,
+    language = 'javascript',
+    canCopy = true,
+    displayHead = true,
+  }: CodeSnippetProps) => {
+    const codeRef = useRef(null)
+    const { translate } = useInternationalization()
+
+    useEffect(() => {
+      if (codeRef && codeRef.current) {
+        Prism.highlightElement(codeRef.current)
+      }
+    })
+
+    return (
+      <div className={tw('code-snippet', 'relative h-full', className)}>
+        {loading ? null : (
```

Comment:
What I've noticed is that if we try to swap similar logic to:

```
{!loading && (
  ...
)}

{loading && (
...
)}
```

we get less indentation & the code starts to look less a pyramid. It also makes it easier to debug since following a couple of `?` can get tricky.